packages/mask/src/social-network-adaptor/twitter.com/collecting/identity.ts

```diff
@@ -60,14 +63,29 @@ function resolveLastRecognizedIdentityInner(
             }
         }
     }
-    const watcher = task.watcher
-        .addListener('onAdd', () => assign())
-        .addListener('onChange', () => assign())
-        .startWatch({
-            childList: true,
-            subtree: true,
+
+    const createWatcher = (selector: LiveSelector<HTMLElement, boolean>) => {
+        const watcher = new MutationObserverWatcher(selector)
+            .addListener('onAdd', () => assign())
+            .addListener('onChange', () => assign())
+            .startWatch({
+                childList: true,
+                subtree: true,
+                attributes: true,
+                attributeFilter: ['src'],
+            })
+
+        window.addEventListener('locationchange', assign, { signal: cancel })
+        cancel.addEventListener('abort', () => {
+            window.removeEventListener('locationchange', assign)
```

Comment:
You don't need to remove it here because you already passed the signal at line 78